### DISCUSSION TITLE
Dealing with split packages in testing dependencies

### DISCUSSION THREAD

**Discussion by TomerFi** [sormuras/junit-platform-maven-plugin#56] [Q&A] [answered]
I wonder...

I have a modular project described with a `module-info.java`,
I'm using `module-info.test` to patch the testing process.

Everything worked well.
That is until I decided to incorporate monolithic test dependencies which happen to introduce some split packages.
https://github.com/mock-server/mockserver/issues/970
The dependencies are of course only needed for testing purposes.

How might one deal with this type of situation?

Is there a way to keep my project modularized but test it as a monolithic?
My first thought was the perhaps removing `module-info.test` will achieve that,
But that only made the plugin throw a `UncheckedIOException` caused by an `AccessDeniedException`.

Of course, removing the project descriptor, `module-info.java` makes my project monolithic and everything works well...
But I would like to keep it modularized.

@sormuras 
Any thoughts on the manner perhaps?

**Comment by sormuras** [comment, marked as answer]
Read and upvoted your ticket at mockserver's - and making this library work on the module path is the way to go.

You could try to repackage all mock server related JAR files into a single one. Then, there's no split package by definition. But every time you need to update the mock server library, you have to repackage it again.

**Comment by TomerFi** [comment]
I ended up replacing mockserver with wiremock.

Wiremock introduced two transitive dependencies with split packages into my code,
Fortunally I don't use either of them, so this configuration worked for me:

```xml
<dependency>
  <groupId>com.github.tomakehurst</groupId>
  <artifactId>wiremock-jre8</artifactId>
  <scope>test</scope>
  <exclusions>
    <exclusion>
      <groupId>com.github.jknack</groupId>
      <artifactId>handlebars</artifactId>
    </exclusion>
    <exclusion>
      <groupId>com.github.jknack</groupId>
      <artifactId>handlebars-helpers</artifactId>
    </exclusion>
  </exclusions>
</dependency>
```

**Comment by sormuras** [comment]
> I have a modular project described with a `module-info.java`, [...]

This? https://github.com/TomerFi/hebcal-api

It's deployment to Maven Central was counted and rated as a "unique" Java module! ;-)

🧩 https://github.com/sormuras/modules/commit/ff9001ff2fe9bc8b99eb8269b062e8186845bead#diff-809b8610e4a13cd7e5869d118ca9f08ea31d456cb0df9bf5eb6702a52f2d41c2R1196

**Reply by TomerFi** [reply]
Yes, that's the one.
My first deployment to Maven Central,
I was wondering if it'll turn up in your modules report.
:-)
